test/test_nn.py

```diff
@@ -6725,6 +6725,21 @@ def get_grid(device='cpu', data=None):
                     gradients = torch.randn_like(out_cpu)
                     out_cpu.backward(gradients)
 
+
+                    # Compare against unvectorized CPU fallback
```

Comment:
nit: Maybe provide a little more context here by writing a note? The reason why we support the unvectorized CPU fallback is because it is used for 64-bit indexing for fp32 inputs so this is a parity test that the unvectorized fallback works as advertised.

Furthermore, it would be nice to refer back to this note in the other changes to `test_grid_sample`. For example, as a code reader, I would be wondering why we're converting `input` and `grid` to float32 in places below (the answer is because the fallback only really gets used for float32)